File: frontend/src/types/alerts.ts
import { z } from 'zod'

export const alertResponseSchema = z.object({
  id: z.number(),
  origin: z.string(),
  destination: z.string(),
  target_price: z.number(),
  notify_channel: z.string(),
  notify_target: z.string(),
  is_active: z.boolean(),
  last_triggered_at: z.string().nullable(),
  created_at: z.string(),
})

export const alertsListResponseSchema = z.object({
  alerts: z.array(alertResponseSchema),
})

export type AlertResponse = z.infer<typeof alertResponseSchema>

export interface AlertCreate {
  origin: string
  destination: string
  target_price: number
  notify_channel: 'webhook'
  notify_target: string
}

File: backend/src/schemas/alerts.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field, HttpUrl


class AlertCreate(BaseModel):
    origin: str = Field(min_length=3, max_length=3, pattern=r"^[A-Z]{3}$")
    destination: str = Field(min_length=3, max_length=3, pattern=r"^[A-Z]{3}$")
    target_price: float = Field(gt=0)
    notify_channel: Literal["webhook"] = "webhook"
    notify_target: HttpUrl  # webhook URL，Pydantic 自动验证格式


class AlertResponse(BaseModel):
    id: int
    origin: str
    destination: str
    target_price: float
    notify_channel: str
    notify_target: str
    is_active: bool
    last_triggered_at: datetime | None
    created_at: datetime


class AlertsListResponse(BaseModel):
    alerts: list[AlertResponse]
